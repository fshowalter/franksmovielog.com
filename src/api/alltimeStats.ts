import type { AlltimeStatsJson } from "./data/alltimeStatsJson";

import { alltimeStatsJson } from "./data/alltimeStatsJson";

export type AlltimeStats = {} & AlltimeStatsJson;

export async function alltimeStats(): Promise<AlltimeStats> {
  return await alltimeStatsJson();
}
