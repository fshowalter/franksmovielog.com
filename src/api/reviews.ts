import rehypeRaw from "rehype-raw";
import rehypeStringify from "rehype-stringify";
import { remark } from "remark";
import remarkGfm from "remark-gfm";
import remarkRehype from "remark-rehype";
import smartypants from "remark-smartypants";
import strip from "strip-markdown";

import type { ReviewedTitleJson } from "./data/reviewedTitlesJson";
import type { MarkdownReview } from "./data/reviewsMarkdown";
import type { MarkdownViewing } from "./data/viewingsMarkdown";

import { allReviewedTitlesJson } from "./data/reviewedTitlesJson";
import { allReviewsMarkdown } from "./data/reviewsMarkdown";
import { allViewingsMarkdown } from "./data/viewingsMarkdown";
import { linkReviewedTitles } from "./utils/linkReviewedTitles";
import { getHtml } from "./utils/markdown/getHtml";
import { removeFootnotes } from "./utils/markdown/removeFootnotes";
import { rootAsSpan } from "./utils/markdown/rootAsSpan";
import { trimToExcerpt } from "./utils/markdown/trimToExcerpt";

let cachedViewingsMarkdown: MarkdownViewing[];
let cachedMarkdownReviews: MarkdownReview[];
let cachedReviewedTitlesJson: ReviewedTitleJson[];
let cachedReviews: Reviews;

if (import.meta.env.MODE !== "development") {
  cachedViewingsMarkdown = allViewingsMarkdown();
  cachedReviewedTitlesJson = await allReviewedTitlesJson();
  cachedMarkdownReviews = allReviewsMarkdown();
}

export type Review = MarkdownReview & ReviewedTitleJson & {};

export type ReviewContent = {
  content: string | undefined;
  excerptPlainText: string;
  viewings: ReviewViewing[];
};

export type ReviewExcerpt = {
  excerpt: string;
};

type Reviews = {
  distinctGenres: string[];
  distinctReleaseYears: string[];
  distinctReviewYears: string[];
  reviews: Review[];
};

type ReviewViewing = MarkdownViewing & {
  mediumNotes: string | undefined;
  venueNotes: string | undefined;
  viewingNotes: string | undefined;
};

export async function allReviews(): Promise<Reviews> {
  if (cachedReviews) {
    return cachedReviews;
  }
  const reviewedTitlesJson =
    cachedReviewedTitlesJson || (await allReviewedTitlesJson());
  const reviews = parseReviewedTitlesJson(reviewedTitlesJson);

  if (!import.meta.env.DEV) {
    cachedReviews = reviews;
  }

  return reviews;
}

export function getContentPlainText(rawContent: string): string {
  return getMastProcessor()
    .use(removeFootnotes)
    .use(strip)
    .processSync(rawContent)
    .toString();
}

export async function loadContent<
  T extends { imdbId: string; rawContent: string; title: string },
>(review: T): Promise<ReviewContent & T> {
  const viewingsMarkdown = cachedViewingsMarkdown || allViewingsMarkdown();
  const reviewedTitlesJson = await allReviewedTitlesJson();

  const excerptPlainText = getMastProcessor()
    .use(removeFootnotes)
    .use(trimToExcerpt)
    .use(strip)
    .processSync(review.rawContent)
    .toString();

  const viewings = viewingsMarkdown
    .filter((viewing) => {
      return viewing.imdbId === review.imdbId;
    })
    .reverse()
    .map((viewing) => {
      return {
        ...viewing,
        mediumNotes: getHtmlAsSpan(viewing.mediumNotesRaw, reviewedTitlesJson),
        venueNotes: getHtmlAsSpan(viewing.venueNotesRaw, reviewedTitlesJson),
        viewingNotes: getHtml(viewing.viewingNotesRaw, reviewedTitlesJson),
      };
    });

  return {
    ...review,
    content: getHtml(review.rawContent, reviewedTitlesJson),
    excerptPlainText,
    viewings,
  };
}

export function loadExcerptHtml<T extends { slug: string }>(
  review: T,
): ReviewExcerpt & T {
  const reviewsMarkdown = cachedMarkdownReviews || allReviewsMarkdown();

  const { rawContent } = reviewsMarkdown.find((markdown) => {
    return markdown.slug === review.slug;
  })!;

  let excerptHtml = getMastProcessor()
    .use(removeFootnotes)
    .use(trimToExcerpt)
    .use(remarkRehype, { allowDangerousHtml: true })
    .use(rehypeRaw)
    .use(rehypeStringify)
    .processSync(rawContent)
    .toString();

  excerptHtml = excerptHtml.replace(/\n+$/, "");
  excerptHtml = excerptHtml.replace(
    /<\/p>$/,
    ` <a class="!no-underline uppercase whitespace-nowrap font-normal font-sans text-accent text-xs  leading-none hover:!underline" href="/reviews/${review.slug}/">Continue reading...</a></p>`,
  );

  return {
    ...review,
    excerpt: excerptHtml,
  };
}

export async function mostRecentReviews(limit: number) {
  const reviewedTitlesJson =
    cachedReviewedTitlesJson || (await allReviewedTitlesJson());

  reviewedTitlesJson.sort((a, b) => b.sequence.localeCompare(a.sequence));
  const slicedTitles = reviewedTitlesJson.slice(0, limit);

  const { reviews } = parseReviewedTitlesJson(slicedTitles);

  return reviews;
}

function getHtmlAsSpan(
  content: string | undefined,
  reviewedTitles: { imdbId: string; slug: string }[],
) {
  if (!content) {
    return;
  }

  const html = getMastProcessor()
    .use(remarkRehype, { allowDangerousHtml: true })
    .use(rehypeRaw)
    .use(rootAsSpan)
    .use(rehypeStringify)
    .processSync(content)
    .toString();

  return linkReviewedTitles(html, reviewedTitles);
}

function getMastProcessor() {
  return remark().use(remarkGfm).use(smartypants);
}

function parseReviewedTitlesJson(
  reviewedTitlesJson: ReviewedTitleJson[],
): Reviews {
  const distinctReviewYears = new Set<string>();
  const distinctReleaseYears = new Set<string>();
  const distinctGenres = new Set<string>();
  const reviewsMarkdown = cachedMarkdownReviews || allReviewsMarkdown();

  const reviews = reviewedTitlesJson.map((title) => {
    for (const genre of title.genres) distinctGenres.add(genre);
    distinctReleaseYears.add(title.year);

    const { date, grade, rawContent } = reviewsMarkdown.find(
      (reviewsmarkdown) => {
        return reviewsmarkdown.slug === title.slug;
      },
    )!;

    distinctReviewYears.add(
      date.toLocaleDateString("en-US", {
        timeZone: "UTC",
        year: "numeric",
      }),
    );

    return {
      ...title,
      date,
      grade,
      rawContent,
    };
  });

  return {
    distinctGenres: [...distinctGenres].toSorted(),
    distinctReleaseYears: [...distinctReleaseYears].toSorted(),
    distinctReviewYears: [...distinctReviewYears].toSorted(),
    reviews,
  };
}
