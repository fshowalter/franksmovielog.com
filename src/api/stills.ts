import { getImage } from "astro:assets";

import { normalizeSources } from "./utils/normalizeSources";

/**
 * Props for still images.
 */
export type StillImageProps = {
  src: string;
  srcSet: string;
};

const images = import.meta.glob<{ default: ImageMetadata }>(
  "/content/assets/stills/*.png",
);

/**
 * Generates OpenGraph still image source URL.
 * @param slug - Identifier for the still image
 * @returns Optimized JPEG image source URL
 */
export async function getOpenGraphStillSrc(slug: string) {
  const stillFilePath = Object.keys(images).find((path) => {
    return path.endsWith(`/${slug}.png`);
  })!;

  const stillFile = await images[stillFilePath]();

  const image = await getImage({
    format: "jpeg",
    height: 675,
    quality: 80,
    src: stillFile.default,
    width: 1200,
  });

  return normalizeSources(image.src);
}

/**
 * Retrieves still image properties for a given slug.
 * @param slug - Identifier for the still image
 * @param options - Image dimensions
 * @param options.height - Desired image height
 * @param options.width - Desired image width
 * @returns Still image properties with src and srcSet
 */
export async function getStillImageProps(
  slug: string,
  {
    height,
    width,
  }: {
    height: number;
    width: number;
  },
): Promise<StillImageProps> {
  const stillFilePath = Object.keys(images).find((path) => {
    return path.endsWith(`/${slug}.png`);
  })!;

  const stillFile = await images[stillFilePath]();

  const optimizedImage = await getImage({
    format: "avif",
    height: height,
    quality: 80,
    src: stillFile.default,
    width: width,
    widths: [0.25, 0.5, 1, 2].map((w) => w * width),
  });

  return {
    src: normalizeSources(optimizedImage.src),
    srcSet: normalizeSources(optimizedImage.srcSet.attribute),
  };
}
