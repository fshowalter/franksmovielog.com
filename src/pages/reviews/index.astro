---
import { getProps } from "~/components/Reviews/getProps";
import { Reviews } from "~/components/Reviews/Reviews";
import Layout from "~/layouts/Layout.astro";

const props = await getProps();
---

<Layout
  article={false}
  canonical={true}
  description={props.deck}
  title="Reviews"
>
  <Reviews client:load {...props} />
</Layout>
