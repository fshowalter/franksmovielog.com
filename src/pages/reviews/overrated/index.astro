---
import React from "react";

import { Backdrop, BreadcrumbLink } from "~/components/Backdrop";
import { Layout as ComponentLayout } from "~/components/Layout";
import { getPropsForOverrated as getProps } from "~/components/Reviews/getProps";
import { Overrated } from "~/components/Reviews/Overrated";
import { ReviewsSubNav } from "~/components/Reviews/ReviewsSubNav";
import Layout from "~/layouts/Layout.astro";

const props = await getProps();

// Separate layout props from interactive props
const { backdropImageProps, deck, ...interactiveProps } = props;

const breadcrumb = React.createElement(BreadcrumbLink, {
  children: "Reviews",
  href: "/reviews/",
});
---

<Layout
  canonical={true}
  meta={{
    description: props.metaDescription,
  }}
  openGraph={{
    description: deck,
    type: "website",
  }}
  title="Overrated Disappointments"
>
  <ComponentLayout addGradient className="bg-subtle">
    <Backdrop
      breadcrumb={breadcrumb}
      deck={deck}
      imageProps={backdropImageProps}
      title="Overrated"
    />
    <ReviewsSubNav active="overrated" />
    <Overrated client:load {...interactiveProps} />
  </ComponentLayout>
</Layout>
