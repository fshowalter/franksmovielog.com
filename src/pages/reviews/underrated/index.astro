---
import { getProps } from "~/components/Underrated/getProps";
import { Underrated } from "~/components/Underrated/Underrated";
import Layout from "~/layouts/Layout.astro";

const props = await getProps();
---

<Layout
  canonical={true}
  meta={{ description: props.metaDescription }}
  openGraph={{ description: props.deck, type: "website" }}
  title="Underrated Surprises"
>
  <Underrated client:load {...props} />
</Layout>
