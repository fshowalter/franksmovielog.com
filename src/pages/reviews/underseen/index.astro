---
import { getProps } from "src/components/Underseen/getProps";
import { Underseen } from "src/components/Underseen/Underseen";
import Layout from "src/layouts/Layout.astro";

const props = await getProps();
---

<Layout title="Underseen Gems" description={props.deck}>
  <Underseen client:load {...props} />
</Layout>
