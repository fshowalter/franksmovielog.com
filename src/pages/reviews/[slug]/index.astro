---
import { allReviews, getContentPlainText } from "~/api/reviews";
import { getProps } from "~/features/review/getProps";
import { Review } from "~/features/review/Review";
import Layout from "~/layouts/Layout.astro";

interface Props {
  contentPlainText: string;
  slug: string;
}

export async function getStaticPaths() {
  const { reviews } = await allReviews();

  return await Promise.all(
    reviews.map(async (review) => {
      return {
        params: {
          slug: review.slug,
        },
        props: {
          contentPlainText: getContentPlainText(review.rawContent),
          slug: review.slug,
        },
      };
    }),
  );
}

const { contentPlainText, slug } = Astro.props;

const props = await getProps(slug);

//trim the string to the maximum length
var description = contentPlainText
  .replaceAll(/\r?\n|\r/g, " ")
  .slice(0, Math.max(0, 160));

//re-trim if we are in the middle of a word
description = description.slice(
  0,
  Math.max(0, Math.min(description.length, description.lastIndexOf(" "))),
);

let title = `${props.value.title} movie review`;

if (props.value.title.length <= 10) {
  title = `${title} & summary`;
}

title = `${title} (${props.value.releaseYear})`;
---

<Layout
  canonical={true}
  meta={{ description }}
  openGraph={{ description, type: "article" }}
  title={title}
>
  <Fragment slot="meta-theme-color">
    <meta
      content="#fff"
      media="(prefers-color-scheme: light)"
      name="theme-color"
    />
    <meta
      content="#322f2f"
      media="(prefers-color-scheme: dark)"
      name="theme-color"
    />
  </Fragment>
  <Review {...props} />
</Layout>
