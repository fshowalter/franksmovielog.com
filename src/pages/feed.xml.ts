import rss from "@astrojs/rss";

import { loadExcerptHtml, mostRecentReviews } from "~/api/reviews";
import { getOpenGraphStillSrc } from "~/api/stills";

const gradeMap: Record<string, string> = {
  A: "&#9733;&#9733;&#9733;&#9733;&#9733;",
  "A+": "&#9733;&#9733;&#9733;&#9733;&#9733;",
  "A-": "&#9733;&#9733;&#9733;&#9733;&#189;",
  B: "&#9733;&#9733;&#9733;&#9733;",
  "B+": "&#9733;&#9733;&#9733;&#9733;",
  "B-": "&#9733;&#9733;&#9733;&#189;",
  C: "&#9733;&#9733;&#9733;",
  "C+": "&#9733;&#9733;&#9733;",
  "C-": "&#9733;&#9733;&#189;",
  D: "&#9733;&#9733;",
  "D+": "&#9733;&#9733;",
  "D-": "&#9733;&#189;",
  F: "&#9733;",
};

export function textStarsForGrade(grade: string) {
  return gradeMap[grade];
}

export async function GET() {
  const reviews = await mostRecentReviews(10);

  const rssItems = await Promise.all(
    reviews.map(async (review) => {
      return await loadExcerptHtml(review);
    }),
  );

  return rss({
    customData:
      "<image><url>https://www.franksmovielog.com/assets/favicon-128.png</url><title>Frank's Movie Log</title><link>https://www.franksmovielog.com/</link></image>",
    // `<description>` field in output xml
    description: "Reviews of current, cult, classic, and forgotten films.",
    // Array of `<item>`s in output xml
    // See "Generating items" section for examples using content collections and glob imports
    items: await Promise.all(
      rssItems.map(async (item) => {
        const stillSrc = await getOpenGraphStillSrc(item.slug);

        return {
          content: `<img src="${stillSrc}" alt=""><p>${textStarsForGrade(
            item.grade,
          )}</p>${item.excerpt}`,
          link: `https://www.franksmovielog.com/reviews/${item.slug}/`,
          pubDate: item.date,
          title: `${item.title} (${item.year})`,
        };
      }),
    ),
    // Pull in your project "site" from the endpoint context
    // https://docs.astro.build/en/reference/api-reference/#contextsite
    site: "https://www.franksmovielog.com",
    // `<title>` field in output xml
    title: "Frank's Movie Log",
  });
}
