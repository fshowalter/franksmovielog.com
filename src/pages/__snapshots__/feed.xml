<?xml version="1.0" encoding="UTF-8" ?>
<rss
  version="2.0"
  xmlns:content="http://purl.org/rss/1.0/modules/content/"
><channel><title>Frank&apos;s Movie Log</title><description
    >Reviews of current, cult, classic, and forgotten films.</description><link
    >https://www.franksmovielog.com/</link><image><url
      >https://www.franksmovielog.com/assets/favicon-128.png</url><title
      >Frank&apos;s Movie Log</title><link
      >https://www.franksmovielog.com/</link></image><item><title
      >The Midnight Meat Train (2008)</title><link
      >https://www.franksmovielog.com/reviews/the-midnight-meat-train-2008/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/the-midnight-meat-train-2008/</guid><pubDate
      >Sat, 27 Apr 2024 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Fthe-midnight-meat-train-2008.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&lt;/p&gt;&lt;p&gt;Sometimes, frustrating films are worse than bad ones. Such is the case with &lt;em&gt;The Midnight Meat Train&lt;/em&gt;. &lt;a href=&quot;/reviews/the-midnight-meat-train-2008/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item><item
    ><title>Horror Express (1972)</title><link
      >https://www.franksmovielog.com/reviews/horror-express-1972/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/horror-express-1972/</guid><pubDate
      >Sat, 27 Apr 2024 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Fhorror-express-1972.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&lt;/p&gt;&lt;p&gt;&lt;em&gt;Horror Express&lt;/em&gt; may be flawed, but the production, performances, and novelty of genre icons Peter Cushing and Christopher Lee working as allies make it hard to resist. &lt;a href=&quot;/reviews/horror-express-1972/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item><item
    ><title>Night Train to Terror (1985)</title><link
      >https://www.franksmovielog.com/reviews/night-train-to-terror-1985/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/night-train-to-terror-1985/</guid><pubDate
      >Sat, 27 Apr 2024 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Fnight-train-to-terror-1985.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&amp;#9733;&lt;/p&gt;&lt;p&gt;For an anthology comprising three unrelated low-budget horror features cut to half or a third of their running time and book-ended by God and Satan riding a train to Las Vegas with an 80s teen pop-rock band, &lt;em&gt;Night Train to Terror&lt;/em&gt; is better than you’d expect. &lt;a href=&quot;/reviews/night-train-to-terror-1985/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item><item
    ><title>Human Lanterns (1982)</title><link
      >https://www.franksmovielog.com/reviews/human-lanterns-1982/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/human-lanterns-1982/</guid><pubDate
      >Wed, 24 Apr 2024 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Fhuman-lanterns-1982.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&amp;#9733;&lt;/p&gt;&lt;p&gt;&lt;em&gt;Human Lanterns&lt;/em&gt; may not be a great martial arts movie or a great horror movie, but it proves an intriguing bridge for Shaw Brothers productions. A transitional piece between the straightforward kung-fu actioners like &lt;span data-imdb-id=&quot;tt0077559&quot;&gt;&lt;em&gt;The Five Deadly Venoms&lt;/em&gt;&lt;/span&gt; that launched the studio to worldwide recognition, and the more overt horrors like &lt;span data-imdb-id=&quot;tt0085951&quot;&gt;&lt;em&gt;The Boxer’s Omen&lt;/em&gt;&lt;/span&gt; that saw the studio trying to adjust to changing audience tastes. &lt;a href=&quot;/reviews/human-lanterns-1982/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item><item
    ><title>Frankenhooker (1990)</title><link
      >https://www.franksmovielog.com/reviews/frankenhooker-1990/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/frankenhooker-1990/</guid><pubDate
      >Thu, 29 Feb 2024 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Ffrankenhooker-1990.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&amp;#9733;&amp;#189;&lt;/p&gt;&lt;p&gt;Don’t let the title fool you. Director and co-writer Frank Henenlotter’s story of a man’s misguided attempt to reanimate his deceased fiancée by attaching her severed head to a sex worker’s body sports a surprising streak of feminist and trans-positive satire beneath its gory black comedy. &lt;strong&gt;Spoilers follow&lt;/strong&gt;. &lt;a href=&quot;/reviews/frankenhooker-1990/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item><item
    ><title>The Entity (1982)</title><link
      >https://www.franksmovielog.com/reviews/the-entity-1982/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/the-entity-1982/</guid><pubDate
      >Sun, 18 Feb 2024 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Fthe-entity-1982.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&lt;/p&gt;&lt;p&gt;Barbara Hershey plays a single mother who finds herself under repeated, violent sexual assault from an unseen force. Everywhere she turns for help, she finds men eager to dismiss, manipulate, or abandon her. &lt;a href=&quot;/reviews/the-entity-1982/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item><item
    ><title>Friday the 13th: The New Blood (1988)</title><link
      >https://www.franksmovielog.com/reviews/friday-the-13th-the-new-blood-1988/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/friday-the-13th-the-new-blood-1988/</guid><pubDate
      >Fri, 13 Oct 2023 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Ffriday-the-13th-the-new-blood-1988.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&lt;/p&gt;&lt;p&gt;I came with an open mind. I’d heard rumblings this Jason meets &lt;span data-imdb-id=&quot;tt0074285&quot;&gt;&lt;em&gt;Carrie&lt;/em&gt;&lt;/span&gt; slasher was bad, but rumblings can be wrong. &lt;a href=&quot;/reviews/friday-the-13th-the-new-blood-1988/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item><item
    ><title>The Night They Raided Minsky&apos;s (1968)</title><link
      >https://www.franksmovielog.com/reviews/the-night-they-raided-minskys-1968/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/the-night-they-raided-minskys-1968/</guid><pubDate
      >Mon, 25 Sep 2023 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Fthe-night-they-raided-minskys-1968.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&lt;/p&gt;&lt;p&gt;For his second feature, director William Friedkin tries another musical comedy, this one about the waning days of burlesque. He should have known better, but given the circumstances, I understand. &lt;a href=&quot;/reviews/the-night-they-raided-minskys-1968/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item><item
    ><title>Night of the Demons (1988)</title><link
      >https://www.franksmovielog.com/reviews/night-of-the-demons-1988/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/night-of-the-demons-1988/</guid><pubDate
      >Thu, 21 Sep 2023 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Fnight-of-the-demons-1988.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&amp;#189;&lt;/p&gt;&lt;p&gt;High-school teens stage a Halloween party in an abandoned funeral home only to fall prey to demons who murder them and possess their corpses. Remind you of anything? &lt;a href=&quot;/reviews/night-of-the-demons-1988/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item><item
    ><title>Aqua Teen Forever: Plantasm (2022)</title><link
      >https://www.franksmovielog.com/reviews/aqua-teen-forever-plantasm-2022/</link><guid
        isPermaLink="true"
      >https://www.franksmovielog.com/reviews/aqua-teen-forever-plantasm-2022/</guid><pubDate
      >Thu, 31 Aug 2023 00:00:00 GMT</pubDate><content:encoded
      >&lt;img src=&quot;/_image/?href=%2Fcontent%2Fassets%2Fstills%2Faqua-teen-forever-plantasm-2022.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=jpeg&quot; alt=&quot;&quot;&gt;&lt;p&gt;&amp;#9733;&amp;#9733;&amp;#9733;&lt;/p&gt;&lt;p&gt;For better and worse, it plays like an extended episode of the series. &lt;a href=&quot;/reviews/aqua-teen-forever-plantasm-2022/&quot;&gt;Read more...&lt;/a&gt;&lt;/p&gt;</content:encoded></item></channel></rss>
